### PR TITLE
[New Rule] AWS STS AssumeRole with New MFA Device [Rule Tuning] AWS IAM Deactivation of MFA Device

### PR DESCRIPTION
**Pull request by imays11**
# Pull Request

*Issue link(s)*: 
- https://github.com/elastic/ia-trade-team/issues/458 
- https://github.com/elastic/ia-trade-team/issues/347

## Summary - What I changed
This PR creates a new rule to identify when a user has assumed a role using a new MFA device. Users can assume a role to obtain temporary credentials and access AWS resources using the AssumeRole API of AWS Security Token Service (STS). 
While a new MFA device is not always indicative of malicious behavior it should be verified as adversaries can use this technique for persistence and privilege escalation.

This PR also adds the proper MITRE technique to existing rule "AWS IAM Deactivation of MFA Device"

## How To Test

<details><summary>Screenshot of working query</summary>
<p>

AssumeRole with new MFA device

</p>
</details> 

You can use the Cloudgoat Scenario: [iam_privesc_by_key_rotation](https://github.com/RhinoSecurityLabs/cloudgoat/blob/d5863b80afd082d853f2e8df1955c6393695a4da/scenarios/iam_privesc_by_key_rotation/README.md) to test this rule. Part of the attack chain is for a user to add a new MFA device to a compromised user before assuming a role for privilege escalation. 

## Checklist

- [x] Added a label for the type of pr: `bug`, `enhancement`, `schema`, `Rule: New`, `Rule: Deprecation`, `Rule: Tuning`, `Hunt: New`, or `Hunt: Tuning` so guidelines can be generated
- [x] Secret and sensitive material has been managed correctly

### CODE REVIEW

**Comment by protectionsmachine** [comment]
## Rule: New - Guidelines

These guidelines serve as a reminder set of considerations when proposing a new rule.

### Documentation and Context

- [ ] Detailed description of the rule.
- [ ] List any new fields required in ECS/data sources.
- [ ] Link related issues or PRs.
- [ ] Include references.

### Rule Metadata Checks

- [ ] `creation_date` matches the date of creation PR initially merged.
- [ ] `min_stack_version` should support the widest stack versions.
- [ ] `name` and `description` should be descriptive and not include typos.
- [ ] `query` should be inclusive, not overly exclusive, considering performance for diverse environments. Non ecs fields should be added to `non-ecs-schema.json` if not available in an integration.
- [ ] `min_stack_comments` and `min_stack_version` should be included if the rule is only compatible starting from a specific stack version.
- [ ] `index` pattern should be neither too specific nor too vague, ensuring it accurately matches the relevant data stream (e.g., use logs-endpoint.process-* for process data).
- [ ] `integration` should align with the `index`. If the integration is newly introduced, ensure the manifest, schemas, and `new_rule.yaml` template are updated.
- [ ] `setup` should include the necessary steps to configure the integration.
- [ ] `note` should include any additional information (e.g. Triage and analysis investigation guides, timeline templates).
- [ ] `tags` should be relevant to the threat and align/added to the `EXPECTED_RULE_TAGS` in the definitions.py file.
- [ ] `threat`, `techniques`, and `subtechniques` should map to ATT&CK always if possible.

#### New BBR Rules
- [ ] `building_block_type` should be included if the rule is a building block and the rule should be located in the `rules_building_block` folder.
- [ ] `bypass_bbr_timing` should be included if adding custom lookback timing to the rule.

### Testing and Validation

- [ ] Provide evidence of testing and detecting the expected threat.
- [ ] Check for existence of coverage to prevent duplication.

**Review by terrancedejesus — APPROVED** [review]

File: rules/integrations/aws/persistence_sts_assume_role_with_new_mfa.toml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    "Tactic: Persistence",
+    "Tactic: Lateral Movement",
+]
+timestamp_override = "event.ingested"
+type = "new_terms"
+
+query = '''
+event.dataset:aws.cloudtrail
+    and event.provider:sts.amazonaws.com
+    and event.action:AssumeRole*
```
Comment (Aegrah) [inline rules/integrations/aws/persistence_sts_assume_role_with_new_mfa.toml:47]: Why do we use a wildcard in an `event.action` statement?

**Review by Aegrah — APPROVED** [review]

File: rules/integrations/aws/persistence_sts_assume_role_with_new_mfa.toml (hunk 1 above)
Comment (imays11) [reply rules/integrations/aws/persistence_sts_assume_role_with_new_mfa.toml:47]: @Aegrah to capture `AssumeRole`, `AssumeRoleWithSAML` and `AssumeRoleWithWebIdentity` I could just explicitly call these out and change to an OR statement if that would be better

Comment (Aegrah) [reply rules/integrations/aws/persistence_sts_assume_role_with_new_mfa.toml:47]: Then I would suggest adding `event.action:(AssumeRole or AssumeRoleWithSAML or AssumeRoleWithWebIdentity)` to enhance rule performance.

Comment (Aegrah) [inline rules/integrations/aws/persistence_sts_assume_role_with_new_mfa.toml:47]:
```suggestion
    and event.action:(AssumeRole or AssumeRoleWithSAML or AssumeRoleWithWebIdentity)
```

**Pull request merged**